src/components/LeftPanelWidget.tsx

```diff
@@ -272,7 +272,12 @@ export class KubeflowKaleLeftPanel extends React.Component<IProps, IState> {
       metadata: { ...this.state.metadata, pipeline_description: desc },
     });
   updateDockerImage = (name: string) =>
-    this.setState({ metadata: { ...this.state.metadata, docker_image: name } });
+    this.setState({
+      metadata: {
+        ...this.state.metadata,
+        docker_image: name || DefaultState.metadata.docker_image,
+      },
+    });
```

Comment:
I suggest we drop that change and perform a replacement (if needed) once the button is pressed.
Leaving it as is would still produce inconsistent behavior when deleting the text of the field.
Since there's a placeholder, users will always know what happens when the field is empty.

Once the button is pressed we now deepcopy notebook's metadata.
If docker image [in that copy] is empty, replace it with `DefaultState.metadata.docker_image`.